src/ui/public/share/views/share.html

```diff
@@ -1,15 +1,140 @@
+          type="text"
+          readonly
+          value="{{share.originalUrl}}"
+        />
+      </div>
+    </div>
   </div>
-  <div class="form-group">
-    <label>
-      Share a link
-    </label>
-    <share-object-url share-as-embed="false"></share-object-url>
+
+  <!-- Right panel -->
+  <div class="share-panel share-panel--right">
+    <!-- Title -->
+    <div class="share-panel-title">
+      Share Snapshot
+    </div>
+
+    <!-- Help text -->
+    <div class="share-panel-help-text">
+      Snapshot URLs encode the current state of the {{share.objectType}} in the URL itself. Edits to the saved {{share.objectType}} won't be visible via this URL.
+    </div>
+
+    <!-- iframe -->
+    <div class="share-panel-section" ng-if="share.allowEmbed">
+      <!-- Header -->
+      <div class="share-panel-header">
+        <div class="share-panel-header__label">
+          Embedded iframe
+        </div>
+        <div class="share-panel-header__actions">
+          <a
+            class="share-panel-header__action"
+            ng-if="!share.shortSnapshotIframeUrl"
+            ng-click="share.useShortUrl('shortSnapshotIframeUrl', share.getEmbeddableUrl(share.snapshotUrl))"
+          >
+            Shorten URL
+          </a>
+
+          <a
+            class="share-panel-header__action"
+            ng-click="share.copyToClipboard('#snapshotIframeUrl')"
+          >
+            Copy
+          </a>
+        </div>
+      </div>
+
+      <!-- Input -->
+      <input
+        id="snapshotIframeUrl"
+        class="share-panel-input"
+        type="text"
+        readonly
+        value="{{share.getIframeUrl(share.shortSnapshotIframeUrl || share.snapshotUrl)}}"
+      />
+
+      <!-- Notes -->
+      <div class="share-panel-form-note">
+        Add to your HTML source. Note that all clients must be able to access Kibana.
+      </div>
+    </div>
+
+    <!-- Link -->
+    <div class="share-panel-section">
+      <!-- Header -->
+      <div class="share-panel-header">
+        <div class="share-panel-header__label">
+          Link
+        </div>
+        <div class="share-panel-header__actions">
+          <a
+            class="share-panel-header__action"
+            ng-if="!share.shortSnapshotUrl"
+            ng-click="share.useShortUrl('shortSnapshotUrl', share.snapshotUrl)"
+          >
+            Shorten URL
+          </a>
+
+          <a
+            class="share-panel-header__action"
+            ng-click="share.copyToClipboard('#snapshotUrl')"
+          >
+            Copy
+          </a>
+        </div>
+      </div>
+
+      <!-- Input -->
+      <input
+        id="snapshotUrl"
+        class="share-panel-input"
+        type="text"
+        readonly
+        value="{{share.shortSnapshotUrl || share.snapshotUrl}}"
+      />
+
+      <!-- Notes -->
+      <div class="share-panel-form-note" ng-if="!share.shortSnapshotUrl">
+        We recommend IE users share a Short URL due to IE’s restrictions on URL length.
```

Comment:
I think this should be more about compatibility with consumers of the URL, rather than sharing it. Maybe something like...

```
We recommend sharing shortened snapshot urls for maximum compatibility. Internet Explorer has URL length restrictions, and some wiki and markup parsers don't do well with the full-length version of the snapshot URL, but the short URL should work great.
```